### ISSUE TITLE
Add ability to inject HttpClient instance

### ISSUE THREAD

**Issue by goldsam** [silentpartnersoftware/Keycloak.Net#28]
Add the ability to inject an HttpClient instance. This library is a no-go for my team without this capability.

**Comment by sps-campbellwray** [comment]
Hi @goldsam, could you please provide some more information? This library is built using [Flurl](https://flurl.dev/) which provides a wrapper for HttpClient, and manages the HttpClient lifecycle, etc. 

It would be a significant design change to move this project from Flurl to a standard HttpClient, so this issue is likely out of scope, unless Flurl has support to configure it with an existing HttpClient.

**Comment by goldsam** [comment]
I understand. Why are you wrapping `HttpClient` this is the platform standard client. Preventing me from injecting my own instance flies in the face of well established conventions. Why did you choose to use an abstraction over an abstraction?

**Comment by sps-campbellwray** [comment]
Hi @goldsam,

I see that you opened an issue on the Flurl project yesterday asking for some more information about the goals of Flurl. I see it has since been deleted, so hopefully you got a satisfactory answer.

As for why it is used in this project, as you can likely see this project was forked from an earlier project, and we took over maintenance of the fork in mid-2023, so I cannot really speak to why it was originally chosen, as Flurl has been a dependency for this project long before we started maintaining it.

With that said, as with many open-source projects we welcome contributions from the community, so, if you would like to be able to inject an existing HttpClient instance via the Factory interface you identified you are welcome to open a Pull Request. Please keep in mind when implementing this change should be backwards compatible with the existing implementation, i.e. it should be optional to inject an HttpClient and your change should not require users to manage their own HttpClient if they do not want to.

If you do open a PR please let me know so that it doesn't get missed,
Thanks.

**Comment by goldsam** [comment]
I've found that it was easier to just [NSwag](https://github.com/RicoSuter/NSwag) to code generate a client from a [generated Keycloak OpenAPI spec](https://github.com/ccouzens/keycloak-openapi). This approach is easier to keep up to date and is considerably more flexible. Thanks though.
